src/microbe_stage/MicrobeAI.cs

```diff
@@ -65,7 +73,13 @@ public void Think(float delta, Random random, MicrobeAICommonData data)
 
         ChooseActions(random, data);
 
-        // Clear the absorbed compounds for run and rumble
+        // Store & clear the absorbed compounds for run and rumble
+        previouslyAbsorbedCompounds.Clear();
+        foreach (var compound in microbe.TotalAbsorbedCompounds.ToList())
```

Comment:
Why is this ToList call here? This unnecessary allocates a duplicate of data. You should work directly on the keypairs in the dictionary.